### PR TITLE
Wake DB before running migrations

### PR DESCRIPTION
**Pull request by revmischa**
Doc: https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.custom_resources-readme.html#custom-resources-for-aws-apis

Can create CDK resources that perform AWS API calls

Screen Shot 2022-08-19 at 5 00 44 PM

### CODE REVIEW

File: stacks/databaseMigrations.ts [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
 import { DbMigrationScript } from './resources/migrationScript';
+import { WakeDB } from './resources/wakeUpDB';
 
 export function DatabaseMigrations({ stack }: StackContext) {
   const net = use(Network);
+  const { cluster } = use(Database);
 
-  new DbMigrationScript(stack, 'MigrationScript', { vpc: net.vpc });
+  // make sure DB is running before applying migration
+  const wakeUp = new WakeDB(stack, 'WakeDB', { cluster });
```
Comment (Mamione) [inline stacks/databaseMigrations.ts:12]: I find it counter intuitive that the wakeUp method doesn't wait for it to wake up and that instead it's delegated to the migration method, couldn't we run a simple inexpensive check command in the wake up method and sleep there?

**Review by Mamione — APPROVED** [review]

File: stacks/databaseMigrations.ts (hunk 1 above)
Comment (revmischa) [reply stacks/databaseMigrations.ts:12]: Yeah that would be better. Requires writing an actual function though. Trying to be lazy here but the retry logic in the migration script was not really ideal so idk.

Comment (Mamione) [reply stacks/databaseMigrations.ts:12]: I mean it doesn't really make a difference it's just how it works internally so not a big deal anyway

**Review by adamszeptycki — APPROVED** [review]

File: stacks/databaseMigrations.ts (hunk 1 above)
Comment (revmischa) [reply stacks/databaseMigrations.ts:12]: I'd like to try improving it one day

**Pull request merged**